### PR TITLE
JPN-624 | Changed code to enable VE Tour for Japanese community

### PR DESCRIPTION
**Pull request by d34th4ck3r**
@per @Wikia/west-wing

### CODE REVIEW

**Comment by perjg** [comment]
LGTM

File: extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js [hunk 1]
```diff
@@ -8,20 +8,21 @@ define('VisualEditorTourExperimentInit',
 			usersWithoutEditExperimentId = 5735670451;
```
Comment (WikiaTech) [inline extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js]: MAJOR Remove the declaration of the unused 'usersWithoutEditExperimentId' variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedVariable)
MAJOR Remove this useless assignment to local variable "usersWithoutEditExperimentId" [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1854)

File: extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentConfig.js [hunk 2]
```diff
@@ -1,64 +1,123 @@
 define('VisualEditorTourExperimentConfig', [], function () {
 	'use strict';
```
Comment (WikiaTech) [inline extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentConfig.js]: MINOR Replace all tab characters in this file by sequences of white-spaces. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATabCharacter)
INFO Use of JavaScript strict mode may result in unexpected behaviour in some browsers. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AStrictMode)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 8 issues
* MAJOR 4 major
* MINOR 2 minor
* INFO 2 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [VisualEditorTourExperimentInit.js#L7](https://github.com/Wikia/app/blob/a5f0e0cb90cea3e41e617e40a90819c94cae16d2/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L7): Remove the declaration of the unused 'freshlyRegisteredExperimentId' variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedVariable)
1. MAJOR [VisualEditorTourExperimentInit.js#L7](https://github.com/Wikia/app/blob/a5f0e0cb90cea3e41e617e40a90819c94cae16d2/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L7): Remove this useless assignment to local variable "freshlyRegisteredExperimentId" [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1854)
1. MINOR [VisualEditorTourExperimentInit.js#L2](https://github.com/Wikia/app/blob/a5f0e0cb90cea3e41e617e40a90819c94cae16d2/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L2): Replace all tab characters in this file by sequences of white-spaces. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATabCharacter)
1. INFO [VisualEditorTourExperimentInit.js#L4](https://github.com/Wikia/app/blob/a5f0e0cb90cea3e41e617e40a90819c94cae16d2/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L4): Use of JavaScript strict mode may result in unexpected behaviour in some browsers. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AStrictMode)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 10 issues
* MAJOR 4 major
* MINOR 3 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [VisualEditorTourExperimentInit.js#L7](https://github.com/Wikia/app/blob/8052047dd1d20c1e49886bcc1a7ddc6d80608a73/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L7): Remove the declaration of the unused 'freshlyRegisteredExperimentId' variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedVariable)
1. MAJOR [VisualEditorTourExperimentInit.js#L7](https://github.com/Wikia/app/blob/8052047dd1d20c1e49886bcc1a7ddc6d80608a73/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L7): Remove this useless assignment to local variable "freshlyRegisteredExperimentId" [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1854)
1. MINOR [VisualEditorTourExperimentInit.js#L2](https://github.com/Wikia/app/blob/8052047dd1d20c1e49886bcc1a7ddc6d80608a73/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L2): Replace all tab characters in this file by sequences of white-spaces. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATabCharacter)
1. MINOR [VisualEditorTourExperimentInitArticleEntry.js#L2](https://github.com/Wikia/app/blob/8052047dd1d20c1e49886bcc1a7ddc6d80608a73/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js#L2): Replace all tab characters in this file by sequences of white-spaces. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATabCharacter)
1. INFO [VisualEditorTourExperimentInit.js#L4](https://github.com/Wikia/app/blob/8052047dd1d20c1e49886bcc1a7ddc6d80608a73/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L4): Use of JavaScript strict mode may result in unexpected behaviour in some browsers. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AStrictMode)
1. INFO [VisualEditorTourExperimentInitArticleEntry.js#L3](https://github.com/Wikia/app/blob/8052047dd1d20c1e49886bcc1a7ddc6d80608a73/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js#L3): Use of JavaScript strict mode may result in unexpected behaviour in some browsers. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AStrictMode)

File: extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js [hunk 3]
```diff
@@ -3,25 +3,24 @@ define('VisualEditorTourExperimentInit',
 	function ($, VETour, veTourConfig, abTest, tracker) {
 		'use strict';
```
Comment (WikiaTech) [inline extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js]: INFO Use of JavaScript strict mode may result in unexpected behaviour in some browsers. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AStrictMode)

File: extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js [hunk 4]
```diff
@@ -3,25 +3,24 @@ define('VisualEditorTourExperimentInit',
...
 
-		var experimentName = 'contribution-experiments',
-			freshlyRegisteredExperimentId = 5654433460,
-			usersWithoutEditExperimentId = 5735670451;
-
+		var experimentName = 'contribution-experiments';
+
```
Comment (WikiaTech) [inline extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* MINOR 4 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [VisualEditorTourExperimentInit.js#L2](https://github.com/Wikia/app/blob/9a29bb18223c7427134b7fe5899508981fec02c3/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L2): Replace all tab characters in this file by sequences of white-spaces. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATabCharacter)
1. MINOR [VisualEditorTourExperimentInitArticleEntry.js#L2](https://github.com/Wikia/app/blob/9a29bb18223c7427134b7fe5899508981fec02c3/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js#L2): Replace all tab characters in this file by sequences of white-spaces. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATabCharacter)
1. INFO [VisualEditorTourExperimentInitArticleEntry.js#L3](https://github.com/Wikia/app/blob/9a29bb18223c7427134b7fe5899508981fec02c3/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js#L3): Use of JavaScript strict mode may result in unexpected behaviour in some browsers. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AStrictMode)

File: extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js [hunk 5]
```diff
@@ -14,9 +14,9 @@ require(['jquery', 'mw', 'VisualEditorTourExperimentInit', 'ext.wikia.spitfires.
 				$editButton.popover({
 					content: '<div class="ve-tour-experiment">' +
 						'<div class="close"></div>' +
-						'<h2>Enter edit mode</h2>' +
-						'<p>Click on the Edit button to go into edit mode - there you will be able to add content ' +
-						'and media to an article page.</p>' +
+						'<h2>編集モードを開く</h2>' +
+						'<p>「編集」ボタンをクリックすると編集モードを開始し、記事が掲載されているページにコンテンツやメディアを追加する'+
+						'ことができます。</p>' +
```
Comment (perjg) [inline extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js:20]: For the strings that are encoded in the config JS file, you use  mw.config.get('wgUserLanguage'); to decide what language to serve to the user.

If that language is set to 'en', this text will still end up in Japanese.
You may want to add a test for  mw.config.get('wgUserLanguage') === 'ja' and only then display the Japanese here.

**Comment by perjg** [comment]
One comment, otherwise LGTM.

File: extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js [hunk 6]
```diff
@@ -11,18 +11,33 @@ require(['jquery', 'mw', 'VisualEditorTourExperimentInit', 'ext.wikia.spitfires.
 				mw.config.get('wgNamespaceNumber') === 0 &&
 				mw.config.get('wgVisualEditorPreferred')
 			) {
-				$editButton.popover({
-					content: '<div class="ve-tour-experiment">' +
+				if (mw.config.get('wgUserLanguage') == 'ja') {
```
Comment (WikiaTech) [inline extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js]: MAJOR Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 8 issues
* MAJOR 1 major
* MINOR 4 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [VisualEditorTourExperimentInit.js#L2](https://github.com/Wikia/app/blob/8f81e7a9e1944ae7a371a46cdb1fbae4a780a907/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L2): Replace all tab characters in this file by sequences of white-spaces. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATabCharacter)
1. MINOR [VisualEditorTourExperimentInitArticleEntry.js#L2](https://github.com/Wikia/app/blob/8f81e7a9e1944ae7a371a46cdb1fbae4a780a907/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js#L2): Replace all tab characters in this file by sequences of white-spaces. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATabCharacter)
1. INFO [VisualEditorTourExperimentInitArticleEntry.js#L3](https://github.com/Wikia/app/blob/8f81e7a9e1944ae7a371a46cdb1fbae4a780a907/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js#L3): Use of JavaScript strict mode may result in unexpected behaviour in some browsers. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AStrictMode)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues
* MINOR 4 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [VisualEditorTourExperimentInit.js#L2](https://github.com/Wikia/app/blob/5a1e5977c88b68f6512aa9eade0fe72d2f00103a/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js#L2): Replace all tab characters in this file by sequences of white-spaces. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATabCharacter)
1. MINOR [VisualEditorTourExperimentInitArticleEntry.js#L2](https://github.com/Wikia/app/blob/5a1e5977c88b68f6512aa9eade0fe72d2f00103a/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js#L2): Replace all tab characters in this file by sequences of white-spaces. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATabCharacter)
1. INFO [VisualEditorTourExperimentInitArticleEntry.js#L3](https://github.com/Wikia/app/blob/5a1e5977c88b68f6512aa9eade0fe72d2f00103a/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js#L3): Use of JavaScript strict mode may result in unexpected behaviour in some browsers. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AStrictMode)

**Comment by perjg** [comment]
LGTM

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* MINOR 1 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. INFO [VisualEditorTourExperimentInitArticleEntry.js#L3](https://github.com/Wikia/app/blob/99398b9731799b9a5a79409061e2450f8d5a1b69/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInitArticleEntry.js#L3): Use of JavaScript strict mode may result in unexpected behaviour in some browsers. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AStrictMode)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.track.js#L7](https://github.com/Wikia/app/blob/9d0fe0c584fa17c0d0f8284ae9eb97362039093c/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L7): This function has 232 lines, which is greater than the 100 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS138)
1. MAJOR [ve.track.js#L7](https://github.com/Wikia/app/blob/9d0fe0c584fa17c0d0f8284ae9eb97362039093c/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L7): Function has a complexity of 33 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [ve.track.js#L142](https://github.com/Wikia/app/blob/9d0fe0c584fa17c0d0f8284ae9eb97362039093c/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L142): Function has a complexity of 11 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.track.js#L7](https://github.com/Wikia/app/blob/0aff48535c5e62136ae1805a6dd3f465acea0314/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L7): This function has 232 lines, which is greater than the 100 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS138)
1. MAJOR [ve.track.js#L7](https://github.com/Wikia/app/blob/0aff48535c5e62136ae1805a6dd3f465acea0314/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L7): Function has a complexity of 33 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [ve.track.js#L142](https://github.com/Wikia/app/blob/0aff48535c5e62136ae1805a6dd3f465acea0314/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L142): Function has a complexity of 11 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js [hunk 7]
```diff
@@ -3,25 +3,31 @@ define('VisualEditorTourExperimentInit',
...
 		}
 
 		function trackPublish() {
-			if (isExperimentVariation() && (isNewlyregistered() || isUserwithoutedit())) {
+			var enable = isAllowedCommunity();
+			if (mw.config.get('wgUserName') != null) {
+				enable = enable && isUserLanguagePreferenceJapanese() && isUserwithoutedit();
+			}
+			if (enable) {
+				debugger;
```
Comment (WikiaTech) [inline extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperimentInit.js]: CRITICAL Remove this debugger statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ADebuggerStatement)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* CRITICAL 1 critical
* MAJOR 3 major

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.track.js#L7](https://github.com/Wikia/app/blob/b5c2412180a32a714332812e66f85b31637ea924/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L7): This function has 232 lines, which is greater than the 100 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS138)
1. MAJOR [ve.track.js#L7](https://github.com/Wikia/app/blob/b5c2412180a32a714332812e66f85b31637ea924/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L7): Function has a complexity of 33 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [ve.track.js#L142](https://github.com/Wikia/app/blob/b5c2412180a32a714332812e66f85b31637ea924/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L142): Function has a complexity of 11 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.track.js#L7](https://github.com/Wikia/app/blob/c2e625ecdf5774289220ff6657e502fb0e3e14b2/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L7): This function has 232 lines, which is greater than the 100 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS138)
1. MAJOR [ve.track.js#L7](https://github.com/Wikia/app/blob/c2e625ecdf5774289220ff6657e502fb0e3e14b2/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L7): Function has a complexity of 33 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. MAJOR [ve.track.js#L142](https://github.com/Wikia/app/blob/c2e625ecdf5774289220ff6657e502fb0e3e14b2/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L142): Function has a complexity of 11 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)

**Comment by perjg** [comment]
LGTM

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [ve.track.js#L7](https://github.com/Wikia/app/blob/a8976fbf19302bed4bec82e0756f52eb91a19b63/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L7): This function has 232 lines, which is greater than the 100 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS138)
2. MAJOR [ve.track.js#L7](https://github.com/Wikia/app/blob/a8976fbf19302bed4bec82e0756f52eb91a19b63/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L7): Function has a complexity of 33 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
3. MAJOR [ve.track.js#L142](https://github.com/Wikia/app/blob/a8976fbf19302bed4bec82e0756f52eb91a19b63/extensions/VisualEditor/wikia/modules/ve/ve.track.js#L142): Function has a complexity of 11 which is greater than 10 authorized. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AFunctionComplexity)
4. MINOR [VisualEditorTourExperiment.js#L7](https://github.com/Wikia/app/blob/a8976fbf19302bed4bec82e0756f52eb91a19b63/extensions/wikia/VisualEditorTourExperiment/scripts/VisualEditorTourExperiment.js#L7): Rename this 'Tour' function to match the regular expression ^[a-z][a-zA-Z0-9]*$ [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS100)

**Comment by perjg** [comment]
LGTM

**Pull request merged**